libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_api.c

```diff
@@ -938,31 +1067,168 @@ TEST( Provisioning_Unit_API, CreateKeysAndCertificateAPIServerResponseAfterTimeo
 
     _testCreateKeysAndCertificateAPIWithServerResponse( &serverResponse,
                                                         _testProvisioningServerResponseThreadTimeoutMs * 2,
-                                                        &_acceptedResponseCallbackForCreateKeysAndCertificateAPI,
+                                                        &_keysAndCertAcceptedResponseCallback,
                                                         AWS_IOT_PROVISIONING_TIMEOUT );
 }
 
 /*-----------------------------------------------------------*/
 
 /**
- * @brief Tests the behavior of @ref provisioning_function_getpKeysAndCertificate when there is a race condition between
- * the library receiving the server response and the response timeout firing. Even in such a case, the API is expected
- * to process the response and invoke the user callback with the device credentials instead of treating the case as a
- * timeout!*/
-TEST( Provisioning_Unit_API, CreateKeysAndCertificateAPIServerResponseAndTimeoutRaceCondition )
+ * @brief Verifies that the CSR API function returns the appropriate error code
+ * on passing invalid parameters.
+ */
+TEST( Provisioning_Unit_API, CreateCertFromCsrAPIInvalidParameters )
+{
+    AwsIotProvisioningError_t status = AWS_IOT_PROVISIONING_SUCCESS;
+    AwsIotProvisioningCreateCertFromCsrCallbackInfo_t callbackInfo =
+    {
+        .userParam = NULL,
+        .function  = _certFromCsrDummyCallback
+    };
+
+
+    /* Uninitialized MQTT connection. */
+    status = AwsIotProvisioning_CreateCertificateFromCsr( NULL,
+                                                          IOT_MQTT_QOS_0,
+                                                          _testCsrString,
+                                                          sizeof( _testCsrString ),
+                                                          0,
+                                                          &callbackInfo );
+    TEST_ASSERT_EQUAL( AWS_IOT_PROVISIONING_BAD_PARAMETER, status );
+
+    /* Invalid CSR data. */
+    status = AwsIotProvisioning_CreateCertificateFromCsr( _pMqttConnection,
+                                                          IOT_MQTT_QOS_0,
+                                                          NULL,
+                                                          sizeof( _testCsrString ),
+                                                          0,
+                                                          &callbackInfo );
```

Comment:
Should the ```status``` not be verified after this call?

Reply:
Good catch.

Reply:
Added